docs/call-tree.md

```diff
@@ -0,0 +1,353 @@
+# Call Tree
+
+A call tree is a summary representation of sample and stack data. It adds together all of the timings of the different functions, and presents them in a tree depending on how they are called. For instance take the following 3 samples, where the letter represents the function that was being called.
+
+```
+ 1ms  2ms  3ms
+  A    A    A
+  |    |    |
+  v    v    v
+  B    B    B
+  |    |    |
+  v    v    v
+  C    C    H
+  |    |    |
+  v    v    v
+  D    F    I
+  |    |
+  v    v
+  E    G
+```
+
+In this case the profiler took 3 samples, each 1 millisecond apart. This approach doesn't scale when you have thousands of samples, so it's best to summarize this information. The call tree does this by combining areas where functions are shared at the same level of a call stack. Taking the above samples and the corresponding relationship between the stacks would produce the following tree.
+
+```
+              A
+              |
+              v
+              B
+            /   \
+           v     v
+          C       H
+        /   \      \
+       v     v      v
+      D       F     I
+      |       |
+      v       v
+      E       G
+```
+
+This structure makes it much easier to tell where time is being spent in different parts of the application. Here the call tree shows that function `A` is the root of our samples, and all samples were called from there. According to this call tree `B` calls out to two different functions, `C` and `H`. However with just this tree we do not yet know how long each function took. This is where self time and running time come into play.
+
+# Self and running time
+
+With the above graph, we know that the functions at the leaf of the graph were the actual functions that were running when we took the samples. So for function `E`, `G`, and `I`, their self times are 1ms. Their running times will be the same as their self time. Now walking up the stack, respectively the function `D`, `F`, and `H` called those functions. Since we never actually observed those functions directly running (they weren't at the leaves), then the self times for all of those functions would be 0ms. However, because the functions they called *did* in fact run, then the running times would be 1ms each. Walking all the way up to root of the graph we get to function `A`. This node on the call tree would have a self time of 0ms as we never actually directly observed it running, but it includes every other function that we did observe, so the running time would be 3ms. The following is a graph of all the running and self times where the first number is the running time, and the second is the self time. So `A:3,0` would be the function `A` with the running time of 3ms, and the self time of 0ms.
```

Comment:
> (they weren't at the leaves)

I'd say "they weren't at the top of the stack" because leaf is a term that describes a tree, and we can have functions with self time that don't end up being leaves of the call tree.